libs/sdk/assembly/crypto.ts

```diff
@@ -186,33 +246,90 @@ class KeyECC extends Key {
             ret[i] = value[i];
         return new PublicKey(ret);
     }
+
+    getPrivateKey(): PrivateKey {
+        const k = String.UTF8.encode(this.name, true);
+        let value = new Uint8Array(32);
+        let result = export_key_raw(k, 0, value.buffer, value.byteLength);
+        const ret: u8[] = [];
+        if (result < 0)
+            return new PrivateKey(ret); // todo : report error
+        if (result > value.byteLength) {
+            // buffer not big enough, retry with a properly sized one
+            value = new Uint8Array(result);
+            result = export_key_raw(k, 0, value.buffer, value.byteLength);
+            if (result < 0)
+                return new PrivateKey(ret); // todo : report error
+        }
+        for (let i = 0; i < result; ++i)
+            ret[i] = value[i];
+        return new PrivateKey(ret);
+    }
 }
 
 class CryptoECDSA {
 
-    static generateKey(keyName: string = ""): KeyECC {
+    static generateKey(keyName: string = "", extractable = false): KeyECC {
         const key = new KeyECC(keyName);
-        let nameBuf = String.UTF8.encode(key.name, true);
-        generate_signing_key(nameBuf);
+        const nameBuf = String.UTF8.encode(key.name, true);
+
+        const usages = new Uint8Array(2);
```

Comment:
generate_key can only generate a signing/private key for now. 
public key can be derived from it. 
maybe replace usage "4" (verify) by "5" (derive_key)